crm_compassion/models/weekly_demand.py

```diff
@@ -80,19 +80,22 @@ def _compute_period_locked(self):
     @api.depends('week_start_date', 'week_end_date')
     @api.multi
     def _compute_demand_events(self):
-        days_before_event = int(self.env['ir.config_parameter'].get_param(
-            'crm_compassion.days_allocate_before_event'))
         for week in self.filtered('week_start_date').filtered('week_end_date'):
-            start_date = fields.Date.from_string(week.week_start_date) + \
-                timedelta(days=days_before_event)
-            end_date = fields.Date.from_string(week.week_end_date) + \
-                timedelta(days=days_before_event)
             events = self.env['crm.event.compassion'].search([
-                ('start_date', '>=', fields.Date.to_string(start_date)),
-                ('start_date', '<=', fields.Date.to_string(end_date)),
+                ('hold_start_date', '>=', week.week_start_date),
+                ('start_date', '<=', week.week_end_date)
             ])
-            allocate = sum(events.mapped('number_allocate_children'))
-            resupply = allocate - sum(events.mapped('planned_sponsorships'))
+            allocate = 0
+            weeks_before_event = 0
+            for event in events:
+                weeks_before_event = (
+                     fields.Date.from_string(event.hold_start_date) -
```

Comment:
It should be `start_date - hold_start_date` instead of `hold_start_date - start_date`